### PR TITLE
Document Pod affinity namespaceSelector

### PR DESCRIPTION
**Pull request by ahg-g**
ref: https://github.com/kubernetes/enhancements/issues/2249

/assign @alculquicondor

### CODE REVIEW

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (liggitt) [inline content/en/docs/concepts/policy/resource-quotas.md:440]:
```suggestion
Preventing users from using cross-namespace affinity terms might be desired since a pod with anti-affinity constraints can block pods from all other namespaces 
```

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (liggitt) [inline content/en/docs/concepts/policy/resource-quotas.md:483]:
```suggestion
`CrossNamespaceAffinity` scope and a hard limit greater than or equal to the number of pods using those fields.
```

**Review by liggitt — COMMENTED** [review]
a couple nits, lgtm

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (ahg-g) [reply content/en/docs/concepts/policy/resource-quotas.md:483]: done.

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md
Comment (sftim) [inline content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:283]: Please add this feature gate into https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ and provide a brief description.

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md
Comment (sftim) [inline content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:281]: Style nit:
```suggestion
This feature is alpha and disabled by default. You can enable the it by setting the
```

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md
Comment (sftim) [inline content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:274]: Style nit:
```suggestion
#### Namespace selector
```

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (sftim) [inline content/en/docs/concepts/policy/resource-quotas.md:433]: Style nit:
```suggestion
### Cross-namespace pod affinity quota
```

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (sftim) [inline content/en/docs/concepts/policy/resource-quotas.md:438]: Is this quota scope only vaid if you have enabled the relevant feature gate?

**Review by sftim — CHANGES_REQUESTED** [review]
@ahg-g this looks good. Please also add `PodAffinityNamespaceSelector` to the list of feature gates, and provide a brief description.

The other feedback I've written (see inline comments) is nits, that you can address if you'd like ot.

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (tengqm) [inline content/en/docs/concepts/policy/resource-quotas.md:462]: Considering that this paragraph is in the "resource-quotas" page, mentioning "namespaces" and "namespaceSelector" in this context is confusing.

I'll take a look at the `ResourceQuotaConfiguration` and see if we can generate a reference API for it.

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (ahg-g) [reply content/en/docs/concepts/policy/resource-quotas.md]: done

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (ahg-g) [reply content/en/docs/concepts/policy/resource-quotas.md]: done.

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (ahg-g) [reply content/en/docs/concepts/policy/resource-quotas.md]: yes, I added a paragraph at the end.

Comment (ahg-g) [reply content/en/docs/concepts/policy/resource-quotas.md]: done.

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (ahg-g) [reply content/en/docs/concepts/policy/resource-quotas.md]: done

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (ahg-g) [reply content/en/docs/concepts/policy/resource-quotas.md]: sg, let me know if you have any suggestions.

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md
Comment (ahg-g) [reply content/en/docs/concepts/scheduling-eviction/assign-pod-node.md]: done.

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md
Comment (ahg-g) [reply content/en/docs/concepts/scheduling-eviction/assign-pod-node.md]: done.

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md
Comment (ahg-g) [reply content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:283]: updated.

File: content/en/docs/concepts/policy/resource-quotas.md
Comment (alculquicondor) [inline content/en/docs/concepts/policy/resource-quotas.md:486]: enable

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md
Comment (alculquicondor) [inline content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:277]: nit: `matching namespaces`

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md
Comment (alculquicondor) [inline content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:281]: enable

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md
Comment (alculquicondor) [inline content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:279]: if omitted or null, as well as the `namespaces` list, affinity/anti-affinity terms only match the namespace of the pod that defines them.

Comment (ahg-g) [reply content/en/docs/concepts/scheduling-eviction/assign-pod-node.md]: updated

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md
Comment (alculquicondor) [inline content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:280]: end sentence with period.

**Review by alculquicondor — COMMENTED** [review]
looks good for sig-scheduling

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:674]: Shouldn't this be:
```suggestion
- `PodOverhead`: Enable the [PodOverhead](/docs/concepts/scheduling-eviction/pod-overhead/) feature.
```
?

**Review by sftim — CHANGES_REQUESTED** [review]
One thing to fix; see inline suggestion.

I suggest if addressing that you also squash commits @ahg-g as this is otherwise looking OK to merge.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md
Comment (ahg-g) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: ops, this is meant for the line above, fixed.

**Comment by sftim** [comment]
@kubernetes/sig-scheduling-pr-reviews &ndash; please check for technical accuracy and add /lgtm if happy

**Comment by alculquicondor** [comment]
/lgtm

from sig-scheduling

/sig auth

**Comment by ahg-g** [comment]
Jordan already lgtmed for sig auth

**Comment by alculquicondor** [comment]
Correct, but the label was missing :)

**Comment by sftim** [comment]
/approve
/hold

If the code change has landed, this is good to unhold.

**Comment by sftim** [comment]
Thanks @ahg-g and reviewers!

**Pull request merged**
